pkg/dashboards/dashboards.go

```diff
@@ -254,11 +237,68 @@ func getDashboard(id string, myid int64) (map[string]interface{}, error) {
 		return nil, err
 	}
 
+	if err := refreshFolderMetadata(id, detailDashboardInfo, myid); err != nil {
+		log.Warnf("getDashboard: Failed to refresh folder metadata for dashboard %s: %v", id, err)
+	}
+
 	return detailDashboardInfo, nil
 }
 
-func updateDashboard(id string, dName string, dashboardDetails map[string]interface{}, myid int64) error {
+func refreshFolderMetadata(id string, dashboardDetails map[string]interface{}, myid int64) error {
+	if isDefaultDashboard(id) {
+		return nil
+	}
+
+	structure, err := readFolderStructure(myid)
+	if err != nil {
+		log.Errorf("refreshFolderMetadata: failed to read structure: %v", err)
+		return err
+	}
+
+	dashboardItem, exists := structure.Items[id]
+	if !exists {
+		log.Warnf("refreshFolderMetadata: dashboard %s not found in structure", id)
+		return nil
+	}
+
+	folderID := dashboardItem.ParentID
+	currentFolder, exists := structure.Items[folderID]
+
+	if !exists {
+		log.Warnf("refreshFolderMetadata: folder %s not found", folderID)
```

Comment:
%v is better format specifier than %s, since the former would work for any data type